config/config.go

```diff
@@ -166,14 +223,327 @@ func (cs configs) ResolveConfig(f func(configs []interface{}) error) error {
 // The default configuration sources are:
 // * Environment Variables
 // * Shared Configuration and Shared Credentials files.
-func LoadDefaultConfig(cfgs ...Config) (aws.Config, error) {
-	var cfgCpy configs
-	cfgCpy = append(cfgCpy, cfgs...)
+func LoadDefaultConfig(ctx context.Context, optFns ...func(*LoadOptions) error) (aws.Config, error) {
+	var cfg aws.Config
+	options := LoadOptions{}
+
+	for _, optFn := range optFns {
+		optFn(&options)
+	}
 
-	cfgCpy, err := cfgCpy.AppendFromLoaders(defaultLoaders)
+	// assign Load Options to configs
+	var cfgCpy = configs{options}
+
+	cfgCpy, err := cfgCpy.AppendFromLoaders(ctx, defaultLoaders)
+	if err != nil {
+		return aws.Config{}, err
+	}
+
+	cfg, err = cfgCpy.ResolveAWSConfig(ctx, defaultAWSConfigResolvers)
 	if err != nil {
 		return aws.Config{}, err
 	}
 
-	return cfgCpy.ResolveAWSConfig(defaultAWSConfigResolvers)
+	return cfg, nil
+}
+
+// Getter functions to satisfy interfaces
+
+// getRegion returns Region from config's LoadOptions
+func (o LoadOptions) getRegion(context.Context) (string, error) {
+	if len(o.Region) != 0 {
+		return o.Region, nil
+	}
+
+	return o.DefaultRegion, nil
+}
```

Comment:
returning `DefaultRegion` probably is not correct here.  Since region can be provided by other sources such as env and shared config.

Resolving the `DefaultRegion` should be its own explicit action and resolver, or it will stomp on top of other resolvers.

Reply:
In addition, `getRegion` probably should have the same return signature as `getDefaultRegion`. If the bool is needed in one it would be needed in both.

Since its not really needed by `getRegion` and the caller will do a check of `len() == 0` then the bool probably isn't needed by getDefaultRegion.

Thoughts?